src/main/java/com/google/jenkins/plugins/k8sengine/KubernetesEngineBuilder.java

```diff
@@ -180,26 +200,125 @@ public void perform(
     // generate a kubeconfig for the cluster
     KubeConfig kubeConfig = KubeConfig.fromCluster(projectId, cluster);
 
-    // run kubectl apply
-    KubectlWrapper.runKubectlCommand(
+    JenkinsRunContext context =
         new JenkinsRunContext.Builder()
             .workspace(workspace)
             .launcher(launcher)
             .taskListener(listener)
             .run(run)
-            .build(),
+            .build();
+    KubectlWrapper kubectl = new KubectlWrapper(context, kubeConfig);
+    kubectl.runKubectlCommand(
+        context,
         kubeConfig,
         "apply",
-        ImmutableList.<String>of("-f", workspace.child(manifestPattern).getRemote()));
+        ImmutableList.<String>of("-f", workspace.child(manifestPattern).getRemote(), "-o", "yaml"));
+
+    if (verifyDeployments) {
+      verified = verify(kubectl, manifestPattern, workspace, listener.getLogger());
+      if (!verified) {
+        throw new AbortException("Kubernetes objects could not be verified.");
+      }
+    }
 
-    // run the after build step if it exists
-    // NOTE(craigatgoogle): Due to the reflective way this class is created, initializers aren't
-    // run, so we still have to check for null.
     if (afterBuildStep != null) {
       afterBuildStep.perform(kubeConfig, run, workspace, launcher, listener);
     }
   }
 
+  private Set<String> getKindsToVerify() {
+    return ImmutableSet.of("Deployment");
+  }
+
+  /*
+   * verify the application of the supplied manifest objects and return true if it succeeded
+   */
+  private boolean verify(
+      KubectlWrapper kubectl, String manifestPattern, FilePath workspace, PrintStream consoleLogger)
+      throws AbortException, InterruptedException, IOException {
+
+    LOGGER.log(
+        Level.INFO,
+        String.format(
+            "GKE verifying deployment to, projectId: %s cluster: %s zone: %s manifests: %s",
+            projectId, clusterName, zone, workspace.child(manifestPattern)));
+
+    consoleLogger.println(
```

Comment:
Out of curiosity, what prompted the use of the listener's logger as opposed to the one provided by LOGGER?

Reply:
I could potentially split between both and only use the listener's logger at the end.  The point is if the build fails for lack of verification, the next thing the user would want to know is why that happened.  Hence it seemed like that should go into the build log.  Although maybe it's a bit verbose ATM.

Reply:
Further if it takes a little while for the deployment(s) to become live, without seeing the data in the console, the user may think things are stuck.